tools/Google.Cloud.Tools.ReleaseManager/UpdateFromBazelCommand.cs

```diff
@@ -0,0 +1,138 @@
+
+namespace Google.Cloud.Tools.ReleaseManager;
+
+/// <summary>
+/// A class to update parameters in the API catalog based on BUILD.bazel files.
+/// (This is a stopgap measure until we've got a single configuration source.)
+/// </summary>
+public sealed class UpdateFromBazelCommand : CommandBase
+{
+    public UpdateFromBazelCommand() : base("update-from-bazel", "Updates the API catalog based on BUILD.bazel files", "id")
+    {
+    }
+
+    protected override void ExecuteImpl(string[] args)
+    {
+        string id = args[0];
+        var catalog = ApiCatalog.Load();
+
+        var ids = id == "all"
+            ? catalog.Apis.Where(api => api.ProtoPath is string && api.Generator == GeneratorType.Micro).Select(api => api.Id).ToArray()
+            : new[] { id };
+
+        bool anyUpdated = false;
+        foreach (var apiId in ids)
+        {
+            if (Update(catalog[apiId]))
+            {
+                anyUpdated = true;
+                Console.WriteLine($"Modified configuration for {apiId}");
+            }
+        }
+
+        if (anyUpdated)
+        {
+            File.WriteAllText(ApiCatalog.CatalogPath, catalog.FormatJson());
+        }
+    }
+
+    public bool Update(ApiMetadata api)
+    {
+        var root = DirectoryLayout.DetermineRootDirectory();
+        var googleapis = Path.Combine(root, "googleapis");
+
+        var bazelFile = Path.Combine(googleapis, api.ProtoPath, "BUILD.bazel");
+        if (!File.Exists(bazelFile))
+        {
+            throw new UserErrorException($"Expected Bazel file does not exist: '{bazelFile}'");
+        }
+
+        var lines = File.ReadAllLines(bazelFile).ToList();
+        var start = lines.IndexOf("csharp_gapic_library(");
+        if (start == -1)
+        {
+            // This is effectively a warning. (AI Platform has it commented out, for example.)
+            Console.WriteLine($"Bazel file for {api.Id} does not include csharp_gapic_library target.");
+            return false;
+        }
+        var end = lines.IndexOf(")", start);
+        // Second check is just a safety-net, in case the end of the target is formatted badly and we've picked up a later one.
+        if (end == -1 || end - start > 20)
+        {
+            throw new UserErrorException($"Cannot find end of csharp_gapic_library target in Bazel file for {api.Id}.");
+        }
+
+        // This is really pretty horrible, but it will do the job...
+        var properties = lines
+            .Skip(start)
+            .Take(end - start)
+            .Select(line => line.TrimEnd(','))
+            .Where(line => line.Contains('='))
+            .Select(line => line.Split('=', 2))
+            .ToDictionary(bits => bits[0].Trim(), bits => bits[1].Trim('"', ' '));
+
+        bool updated = false;
+
+        bool restNumericEnums = properties.TryGetValue("rest_numeric_enums", out var restNumericEnumsValue) && restNumericEnumsValue == "True";
+        if (api.RestNumericEnums != restNumericEnums)
```

Comment:
Is there a check somewhere that once enabled, numeric enums cannot be disabled?
I know you are checking for True only above, which is fine, but if they were enabled once and are now disabled, our apis.json will be out of sync with the bazel files. I guess that disabling the numeric enums is a breaking change, so it should be caught upstream, just wondering.

Reply:
Nope, if it's true before, but then set to "False" (or *anything* other than "True") then we'll propagate that to the config. (Eventually "true" may become the default, but we can deal with that later...)